### PR TITLE
Tools: Added how to use Gstreamer Debugger (gst-debugger)

### PR DESCRIPTION
**Pull request by leemgs**
Fixed issue 132#issuecomment-439788784.

This commit is to explain how to use a Gstreamer Debugger tool to
introspect a gst-pipeline remotely with a graphical client.

**Changes proposed in this PR:**
* Version 2:
1. Added GStreamer-based totem player as an example

* Version 1:
1. Added How to use Gstreamer debugger
2. Added How to patch ver 0.90.0 to build on Ubuntu 16.04

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

--

### CODE REVIEW

**Comment by leemgs** [comment]
* After this PR: 
image

File: tools/debugging/README.md [hunk 1]
```diff
@@ -57,3 +57,48 @@ $ eog ${number_of_test_case}.png
...
+### gst-debugger
+gst-debugger (a.k.a Gstreamer Debugger) toolset allows to introspect gst-pipeline remotely. It provides graphical client, and GStreamer's plugin.
+This guide is written on Ubuntu 16.04 X86_64 distribution.
+
+#### Build the source code
+```bash
+$ git clone https://github.com/GNOME/gst-debugger.git
+$ cd gst-debugger
+$ git checkout 0.90.0
+$ ./audogen.sh
```
Comment (ohsewon) [inline tools/debugging/README.md]: typo 'audogen' -> 'autogen'

Comment (ohsewon) [reply tools/debugging/README.md]: for './autogen.sh', need 'libprotobuf-c' package.
but, `sudo apt install libprotobuf-c` doesn't work.

Comment (leemgs) [reply tools/debugging/README.md]: Typo is fixed.   If you need meed "checking for ****" statement while running autogen.sh script.  You have to install required package. 

* In my own Ubuntu: 
```bash
 dpkg -l | grep protobuf
ii  libmirprotobuf3:amd64                         0.26.3+16.04.20170605-0ubuntu1.1             amd64        Display server for Ubuntu - RPC definitions
ii  libprotobuf-c-dev                             1.2.1-1                                      amd64        Protocol Buffers C static library and headers (protobuf-c)
ii  libprotobuf-c1                                1.2.1-1                                      amd64        Protocol Buffers C shared library (protobuf-c)
ii  libprotobuf-dev:amd64                         2.6.1-1.3                                    amd64        protocol buffers C++ library (development files)
ii  libprotobuf-lite9v5:amd64                     2.6.1-1.3                                    amd64        protocol buffers C++ library (lite version)
ii  libprotobuf9v5:amd64                          2.6.1-1.3                                    amd64        protocol buffers C++ library
ii  protobuf-c-compiler                           1.2.1-1                                      amd64        Protocol Buffers C compiler (protobuf-c)
ii  protobuf-compiler                             2.6.1-1.3                                    amd64        compiler for protocol buffer definition files
ii  python-protobuf                               2.6.1-1.3                                    amd64        Python bindings for protocol buffers
```

Comment (leemgs) [reply tools/debugging/README.md]: Note: 
> GStreamer's plugin. This guide is written on Ubuntu 16.04 X86_64 distribution.

Comment (ohsewon) [reply tools/debugging/README.md]: What I needed is 'libprotobuf-c-dev'.

**Comment by leemgs** [comment]
**Changes proposed in this PR:**
* Version 2:
1. Added GStreamer-based totem player as an example

* Version 1:
1. Added How to use Gstreamer debugger
2. Added How to patch ver 0.90.0 to build on Ubuntu 16.04

File: tools/debugging/README.md (hunk 1 above)
Comment (ohsewon) [reply tools/debugging/README.md]: Additionally need to install two packages, 'protobuf-compiler' and 'libboost-all-dev'

Comment (leemgs) [reply tools/debugging/README.md]: You describe  why packaging (e.g., .deb, .rpm) is important. :)

**Pull request merged**